velox/exec/Window.cpp

```diff
@@ -628,14 +691,30 @@ RowVectorPtr Window::getOutput() {
     }
   }
 
+  if (!currentPartition_->isComplete() && (currentPartition_->numRows() <= 1)) {
+    // The numRows may be 1, because we keep the last row in previous batch to
+    // compare with the first row in next batch to determine whether they are in
+    // same peer group.
+    return nullptr;
+  }
+
   auto numOutputRows = std::min(numRowsPerOutput_, numRowsLeft);
   auto result = BaseVector::create<RowVector>(
       outputType_, numOutputRows, operatorCtx_->pool());
 
   // Compute the output values of window functions.
   auto numResultRows = callApplyLoop(numOutputRows, result);
-  return numResultRows < numOutputRows
-      ? std::dynamic_pointer_cast<RowVector>(result->slice(0, numResultRows))
+  if (currentPartition_ && currentPartition_->isPartial()) {
```

Comment:
Is this condition really needed ?

Reply:
@aditi-pandit currentPartition_->isPartial() condition is no need. But if (currentPartition_) is needed. Because the currentPartition_  maybe nullptr [here](https://github.com/facebookincubator/velox/blob/main/velox/exec/Window.cpp#L590).

Reply:
By introducing this [check](https://github.com/facebookincubator/velox/pull/9025/files#diff-afebf9a7a0d6dbd1b2c2cb98b36af7a32be8e35a92aa04629db5582519e5003fR57), exception will be thrown for non-partial partitions if we remove the current condition currentPartition_->isPartial().